src/components/fields/MultiSchemaField.js

```diff
@@ -85,7 +89,14 @@ class AnyOfField extends Component {
     const selectedOption = parseInt(event.target.value, 10);
     const { formData, onChange, options } = this.props;
 
-    if (guessType(formData) === "object") {
+    const newOption = options[selectedOption];
+
+    // If the new option is of type object and the current data is an object,
+    // discard properties added using the old option.
+    if (
+      guessType(formData) === "object" &&
+      (newOption.type === "object" || newOption.properties)
+    ) {
```

Comment:
I'm having trouble understanding how this piece of code fixes the issue "should empty the form data when switching from an option of type 'object'." It looks like in that case, `newOption` would not be an object, so then the properties added by the old option would not be deleted?

Reply:
`newOption` is the schema that the form has changed to. When we move into this code path, `formData` will always get set to an object, which is undesirable if the new object is, for example, a string. Previously, if the value of `newOption` was `{ "type": "string" }`, and the `formData` gets set to an object, then the UI would display an input with the text `[Object object]`.

Reply:
Makes sense -- thanks!